src/editorKit.ts

```diff
@@ -170,9 +174,22 @@ export default class EditorKitBase {
 
       this.delegate.setEditorRawText(text)
 
+      if (this.delegate.onNoteContentChange) {
+        const previousContent = previousNote?.content
+        const newContent = this.note?.content
+
+        if (previousContent != newContent) {
```

Comment:
This does reference comparison and not deep value comparison, so will likely always evaluate to true. Only way to check for equality is either recursively iterate until you get to all primitive values, or to JSON.stringify left and right and compare those values. However the latter approach may have a performance cost.